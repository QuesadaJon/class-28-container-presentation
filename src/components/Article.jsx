import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ author, title, description }) => (
  <figure>
    <p>{author}</p>
    <p>{title}</p>
    <p>{description}</p>
  </figure>
);

Article.propTypes = {
  author: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default Article;
