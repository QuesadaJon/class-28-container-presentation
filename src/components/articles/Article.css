.Article {
  text-decoration: none;
  color: black;
  cursor: pointer;
  
  figure {
    width: 14rem;
    text-align: center;
  }
  
  img {
    width: 100%;
  }
}
