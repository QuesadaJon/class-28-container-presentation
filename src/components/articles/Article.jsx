import React from 'react';
import PropTypes from 'prop-types';
import styles from './Article.css';

const Article = ({ author, image, title, description, url }) => (
  <a className={styles.Article} href={url}>
    <figure>
      <img src={image} alt={title} />
      <figcaption>{title}</figcaption>
      <p>{author}</p>
      <p>{description}</p>
    </figure>
  </a>
);

Article.propTypes = {
  author: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default Article;
