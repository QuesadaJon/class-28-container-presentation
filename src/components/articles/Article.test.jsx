import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  it('matches a snapshot', () => {
    const { asFragment } = render(
      <Article 
        author="author"
        image="image"
        title="title"
        description="description"
        url="url"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
