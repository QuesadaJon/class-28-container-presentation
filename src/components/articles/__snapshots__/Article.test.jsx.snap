// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component matches a snapshot 1`] = `
<DocumentFragment>
  <a
    class="Article"
    href="url"
  >
    <figure>
      <img
        alt="title"
        src="image"
      />
      <figcaption>
        title
      </figcaption>
      <p>
        author
      </p>
      <p>
        description
      </p>
    </figure>
  </a>
</DocumentFragment>
`;
